server/README.rst

```diff
@@ -398,3 +398,29 @@ The job_status_webhook parameter is required for this endpoint. Other parameters
   .. code-block:: console
 
     $ curl http://localhost:8000/v1/queues/wait_times?queue=foo\&queue=bar
+
+**[GET] /v1/authenticate/token/<client_id>** - Authenticate client key and return JWT with permissions
```

Comment:
Let's change this slightly. 
- Instead of `/v1/authenticate/token/...`, let's just use `/v1/oauth2/token` to be a bit more consistent with things like hexr (we may also eventually want to add data like grate-type and scope, but it's not clear how we would use that just yet in our current spec).
- I think this would usually be a POST request, not GET
- Instead of specifying the client_id here, let's use something more standard like basic auth. We should *ALWAYS* use this with https, but instead of inventing our own header format here, we can use [RFC3986](https://datatracker.ietf.org/doc/html/rfc3986) compliant basic auth headers which are base64 encoded. So that means the header you should be looking for in the API looks like this:
```
Authorization: Basic BASE_64_ENCODE_OF_USERNAME:PASSWORD
```
When you b64decode that, then you get the original data back as username:password. Along with https of course, this is generally regarded as much better than transmitting it without some sort of encoding. Fun fact, the other thing specified in that RFC, which is adopted by pretty much every client today I think, means that if you do `curl https://myuser:mypassword@testflinger.canonical.com/v1/login` then it will automatically b64encode it for you on the client side! :)

Reply:
Don't forget to update the README here!